src/main/java/mil/nga/tiff/Rasters.java

```diff
@@ -100,34 +120,26 @@ public Rasters(int width, int height, int samplesPerPixel,
 	 *            width of pixels
 	 * @param height
 	 *            height of pixels
-	 * @param samplesPerPixel
-	 *            samples per pixel
-	 * @param bitsPerSample
-	 *            bits per sample
-	 * @param sampleFieldTypes
-	 *            Field type for each sample
+	 * @param sampleTypes
+	 *            Sample type for each sample
 	 * @param sampleValues
 	 *            empty sample values double array
 	 * @param interleaveValues
 	 *            empty interleaved values array
 	 */
-	public Rasters(int width, int height, int samplesPerPixel,
-			List<Integer> bitsPerSample, FieldType[] sampleFieldTypes, ByteBuffer[] sampleValues,
-			ByteBuffer interleaveValues) {
+	public Rasters(int width, int height, SampleType[] sampleTypes, ByteBuffer[] sampleValues,
+					ByteBuffer interleaveValues) {
 		this.width = width;
 		this.height = height;
-		this.samplesPerPixel = samplesPerPixel;
-		this.bitsPerSample = bitsPerSample;
-		this.sampleFieldTypes = sampleFieldTypes;
+		this.sampleTypes = sampleTypes;
 		this.sampleValues = sampleValues;
 		this.interleaveValues = interleaveValues;
 		validateValues();
-		for (int bits : bitsPerSample) {
-			if ((bits % 8) != 0) {
-				throw new TiffException("Sample bit-width of " + bits
-						+ " is not supported");
-			}
-		}
+	}
+
+	public Rasters(int width, int height, int samplesPerPixel, SampleType sampleType) {
+		this(width, height, new SampleType[samplesPerPixel], new ByteBuffer[samplesPerPixel], null);
```

Comment:
Should be this(width, height, new SampleType[samplesPerPixel], ByteOrder.nativeOrder()), otherwise the ByteBuffer does not get allocated. Another suggestion: put new SampleType[samplesPerPixel] and fill into a separate function so that the constructor already gets a filled array in case a check is added to validate() later

Reply:
You're completely right about this.